src/js_parser.zig

```diff
@@ -4982,7 +5007,6 @@ fn NewParser_(
                             .import_record_id = import_record_index,
                         }) catch unreachable;
                         p.import_items_for_namespace.put(p.allocator, namespace_ref, ImportItemForNamespaceMap.init(p.allocator)) catch unreachable;
-                        p.ignoreUsage(p.require_ref);
```

Comment:
why is this and the other ignoreUsage not necessary anymore?

Reply:
i changed alot of how require_ref/__require was tracked.

i'm almost certain that the usage count of the require_ref `the unbound require` global, doesn't actually matter, but the usage count of the runtime.js `__require` symbol does. that's what `p.recordUsageOfRuntimeRequire();` is doing, and it is only called when we are 100% certain the ast node is a runtime require call.

in this path we are returning either namespace_ref (not a runtime require), or e_require_string, which *may* be a runtime require. and in either case an import record has been created.

at the end, we check for these and mark them as needed (see `var runtime_require_uses`)

i want to add tests to bundle each possible way to call require separately to ensure nothing is mistaken. i spent an absurd amount of time dealing with DCE of the __require helper.